Lib/fontTools/ttLib/tables/_k_e_r_n.py

```diff
@@ -115,50 +132,95 @@ def decompile(self, data, ttFont):
 		glyphOrder = ttFont.getGlyphOrder()
 		for k in range(nPairs):
 			left, right, value = next(it), next(it), next(it)
-			if value >= 32768: value -= 65536
+			if value >= 32768:
+				value -= 65536
 			try:
 				kernTable[(glyphOrder[left], glyphOrder[right])] = value
 			except IndexError:
-				# Slower, but will not throw an IndexError on an invalid glyph id.
-				kernTable[(ttFont.getGlyphName(left), ttFont.getGlyphName(right))] = value
-		if len(data) > 6 * nPairs + 4: # Ignore up to 4 bytes excess
-			log.warning("excess data in 'kern' subtable: %d bytes", len(data) - 6 * nPairs)
+				# Slower, but will not throw an IndexError on an invalid
+				# glyph id.
+				kernTable[(
+					ttFont.getGlyphName(left),
+					ttFont.getGlyphName(right))] = value
+		if len(data) > 6 * nPairs + 4:  # Ignore up to 4 bytes excess
+			log.warning(
+				"excess data in 'kern' subtable: %d bytes",
+				len(data) - 6 * nPairs)
 
 	def compile(self, ttFont):
 		nPairs = len(self.kernTable)
 		searchRange, entrySelector, rangeShift = getSearchRange(nPairs, 6)
-		data = struct.pack(">HHHH", nPairs, searchRange, entrySelector, rangeShift)
+		data = struct.pack(
+			">HHHH", nPairs, searchRange, entrySelector, rangeShift)
 
 		# yeehee! (I mean, turn names into indices)
 		try:
 			reverseOrder = ttFont.getReverseGlyphMap()
-			kernTable = sorted((reverseOrder[left], reverseOrder[right], value) for ((left,right),value) in self.kernTable.items())
+			kernTable = sorted(
+				(reverseOrder[left], reverseOrder[right], value)
+				for ((left, right), value) in self.kernTable.items())
 		except KeyError:
 			# Slower, but will not throw KeyError on invalid glyph id.
 			getGlyphID = ttFont.getGlyphID
-			kernTable = sorted((getGlyphID(left), getGlyphID(right), value) for ((left,right),value) in self.kernTable.items())
+			kernTable = sorted(
+				(getGlyphID(left), getGlyphID(right), value)
+				for ((left, right), value) in self.kernTable.items())
 
 		for left, right, value in kernTable:
 			data = data + struct.pack(">HHh", left, right, value)
-		return struct.pack(">HHH", self.version, len(data) + 6, self.coverage) + data
+		# ensure mask bits 8-15 (subtable format) are set to 0
+		self.coverage &= ~0xFF
```

Comment:
here I follow the pattern found elsewhere in ttLib whereby any necessary fix-ups or modifications occur at compile time.